examples/solidity/SimplePonzi/simplePonzi.qnt

```diff
@@ -0,0 +1,109 @@
+// -*- mode: Bluespec; -*-
+/**
+ * A simple example of a Ponzi scheme, inspired by the Solidity contract [1].
+ * Do not use it in production! The purpose of this example is to show how to
+ * specify and check interesting safety and liveness properties with Quint.
+ *
+ * [1]: https://github.com/djokicx/sol-simple-ponzi/blob/master/contracts/SimplePonzi.sol
+ *
+ * Igor Konnov, Informal Systems, 2023
+ */
+module simplePonzi {
+  // addresses are string literals
+  type Addr = str
+
+  // the set of addresses is not fixed, see simplePonziTest below
+  const ADDR: Set[Addr]
+
+  // a state of the EVM that is observed/modified by the contract
+  type EvmState = {
+    // the account balances for every address
+    balances: Addr -> int
+  }
+
+  // a state of the contract
+  type PonziState = {
+   // the address of the last investor, initially of the contract owner
+    currentInvestor: Addr,
+    // the investment made by the current investor
+    currentInvestment: int,
+  }
+
+  // create a new instance of the contract
+  pure def newPonzi(owner: Addr): PonziState = {
+    currentInvestor: owner, currentInvestment: 0
+  }
+
+  // The result of applying a method.
+  // If error != "", then the state are not modified,
+  // and error contains the error message.
+  // Otherwise, evm and contract contain the new states.
+  type Result = {
+    error: str,
+    evm: EvmState,
+    ponzi: PonziState,
+  }
+
+  // Receive an investment and distribute the rewards (to the previous investor).
+  pure def receive(evm: EvmState,
+      ponzi: PonziState, investor: Addr, amount: int): Result = {
+    if (amount > evm.balances.get(investor)) {
+      error: "Insufficient funds",
+      evm: evm,
+      ponzi: ponzi,
+     } else if (amount < 11 * ponzi.currentInvestment / 10) {
```

Comment:
```suggestion
     } else if (amount * 10 < 11 * ponzi.currentInvestment) {
```

Reply:
Prevent any weird behavior from rounding

Reply:
What weird behavior? Maybe that's what we should talk about during the presentation!